### PR TITLE
[TTNN] Support four-value padding for Conv2d

### PR DESCRIPTION
**Pull request by jserbedzijaTT**
### Ticket
[Link to Github Issue](https://github.com/tenstorrent/tt-mlir/issues/2822)

### What's changed
- Added support for passing 4 values to the padding attribute of Conv2dOp
- Refactored verification logic for `ttir.Conv2d` and `ttnn.Conv2d`

### CODE REVIEW

File: include/ttmlir/OpModel/TTNN/Conversion.h [hunk 1]
```diff
@@ -3,6 +3,7 @@
 // SPDX-License-Identifier: Apache-2.0
 #ifndef TTMLIR_OPMODEL_TTNN_CONVERSION_H
 #define TTMLIR_OPMODEL_TTNN_CONVERSION_H
+#include "llvm/Support/raw_ostream.h"
```
Comment (sdjordjevicTT) [inline include/ttmlir/OpModel/TTNN/Conversion.h:6]: This include is probably not needed?

File: lib/Dialect/TTIR/IR/TTIROps.cpp [hunk 2]
```diff
@@ -428,193 +428,401 @@ mlir::tt::ttir::GetDimensionSizeOp::fold(FoldAdaptor adaptor) {
...
+enum OutputDim : unsigned {
+  OUTPUT_BATCH = 0,
+  OUTPUT_HEIGHT = 1,
+  OUTPUT_WIDTH = 2,
+  OUTPUT_CHANNEL = 3
+};
 
-  if (inputType.getRank() != 4) {
-    return emitOpError("Input must be a 4D tensor");
+// If the input and output tensors are flattened, this is the dimension upon
```
Comment (sdjordjevicTT) [inline lib/Dialect/TTIR/IR/TTIROps.cpp:446]: Consider moving this into a utils file...

File: lib/Dialect/TTNN/IR/TTNNOps.cpp [hunk 3]
```diff
@@ -160,167 +161,381 @@ ::mlir::LogicalResult mlir::tt::ttnn::PrepareConv2dWeightsOp::verify() {
...
-  mlir::RankedTensorType inputType = getInput().getType();
-  mlir::RankedTensorType weightType = getWeight().getType();
-  mlir::RankedTensorType outputType = getResult().getType();
-  std::optional<mlir::RankedTensorType> bias =
-      getBias().getImpl() ? std::make_optional(getBias().getType())
-                          : std::nullopt;
-
-  if (inputType.getRank() != 4) {
-    return emitOpError("Input must be a 4D tensor");
+namespace Conv2dVerification {
```
Comment (sdjordjevicTT) [inline lib/Dialect/TTNN/IR/TTNNOps.cpp:164]: same here

**Review by sdjordjevicTT — APPROVED** [review]
A couple of nits inline. Thanks!
